### PR TITLE
add Node attributes and RunList to the Node struct

### PR DESCRIPTION
**Pull request by spheromak**
- updated node.json with more data
- added tests
- removed un needed alias methods as we can acces the embedded types

### CODE REVIEW

File: chef/node.go [hunk 1]
```diff
@@ -7,22 +7,26 @@ import (
...
 type Node struct {
 	*Reader
+	*NativeNode
 }
 
+type RunList []string
+
 // NativeNode represents the native Go version of the deserialized Node type
-// BUG(fujin): Add other destructing fields like run_list, override attributes to NativeNode
 type NativeNode struct {
```
Comment (bradbeam) [inline chef/node.go]: Should we also add json_class and chef_type?

Comment (spheromak) [inline chef/node.go]: IMO, No AFAIK neither are required by chef, tho they may come in from chef.

Comment (bradbeam) [inline chef/node.go]: What about chef_environment?

**Comment by spheromak** [comment]
@fujin Merge this thing

**Comment by fujin** [comment]
@spheromak can you squash rebase it up into one commit?

**Pull request merged**